app/com/arpnetworking/metrics/portal/alerts/impl/DailyPartitionCreator.java

```diff
@@ -131,49 +135,59 @@ public static Props props(
     }
 
     /**
-     * Ask the actor referenced by {@code ref} to start execution.
-     * <p>
-     * This will execute partition creation exactly once and then periodically thereafter using the actor's props.
+     * Ask the actor referenced by {@code ref} to stop execution.
      *
      * @param ref an {@code DailyPartitionCreator}.
      * @param timeout timeout for the operation
      * @throws ExecutionException if an exception was thrown during execution.
-     * @throws InterruptedException if the actor does not reply within the allotted timeout, or if the actor thread was
+     * @throws InterruptedException if the actor does not stop within the allotted timeout, or if the actor thread was
      * interrupted for other reasons.
      */
-    public static void start(
-            final ActorRef ref,
-            final Duration timeout
-    ) throws ExecutionException, InterruptedException {
-        Patterns.ask(
-                ref,
-                START_TICKING,
-                timeout
-        )
-                .thenCompose(reply -> {
-                    @SuppressWarnings("unchecked") final Optional<Throwable> o = (Optional<Throwable>) reply;
-                    final CompletableFuture<Void> future = new CompletableFuture<>();
-                    o.ifPresent(future::completeExceptionally);
-                    future.complete(null);
-                    return future;
-                })
-                .toCompletableFuture()
-                .get();
+    public static void stop(final ActorRef ref, final Duration timeout) throws ExecutionException, InterruptedException {
+        Patterns.gracefulStop(ref, timeout).toCompletableFuture().get();
     }
 
     /**
-     * Ask the actor referenced by {@code ref} to stop execution.
+     * Ask the actor referenced by {@code ref} to create the partition(s) needed
+     * for the given date.
      *
      * @param ref an {@code DailyPartitionCreator}.
+     * @param date The partition date
      * @param timeout timeout for the operation
      * @throws ExecutionException if an exception was thrown during execution.
-     * @throws InterruptedException if the actor does not stop within the allotted timeout, or if the actor thread was
+     * @throws InterruptedException if the actor does not respond within the allotted timeout, or if the actor thread was
      * interrupted for other reasons.
      */
-    public static void stop(final ActorRef ref, final Duration timeout) throws ExecutionException, InterruptedException {
-        Patterns.gracefulStop(ref, timeout).toCompletableFuture().get();
+    public static void ensurePartitionExistsForDate(
```

Comment:
Not entirely without merit, and not as concerned, but at least may be simpler with `PatternsCS`. I believe we also doubled up on the timeouts with this pattern in MAD and CAGG; e.g.

https://github.com/ArpNetworking/metrics-aggregator-daemon/commit/158884ec04cc53f5b7b6dc705e78cd1ffab437c4#diff-fed66f6be12dfd97b6fb62212dd0d6ebR52

Reply:
As mentioned above PatternsCS and Patterns are essentially the same API for this version of Akka - but I'll add the timeouts. I'm pretty sure I'm the one that asked you to double up on that code to begin with 😄